File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.java
// JAVA code to find Largest_common_multiple of n elements of the array

import java.util.Scanner;
import java.util.Arrays;

class LCM 
{
    public static void main(String args[]) 
    {
        int n;
        Scanner s = new Scanner(System.in);
        System.out.println("Enter no. of elements in array:");
        n = s.nextInt();
        int a[] = new int[n];
        System.out.println("Enter all the elements:");
        for (int i = 0; i < n; i++) 
        {
            // Taking input from the user
            a[i] = s.nextInt(); 
        }
        int lcm = 1;
        for (int each: a) 
        {
            lcm = calculateLcm(lcm, each);
        }
        System.out.println("LCM for " + Arrays.toString(a) + " is : " + lcm);
    }

    // Calculating LCM by using GCD
    private static int calculateLcm(int lcm, int each) 
    {
        return lcm * each / gcd(lcm, each);
    }

    // Calculating GCD
    private static int gcd(int val1, int val2) 
    {
        if (val1 == 0 || val2 == 0)
            return 0;

        if (val1 == val2)
            return val1;

        if (val1 > val2)
            return gcd(val1 - val2, val2);
        return gcd(val1, val2 - val1);
    }
}

/*Enter no. of elements in array:
6
Enter all the elements:
1 2 3 4 5 6
LCM for [1, 2, 3, 4, 5, 6] is : 60*/

File: Largest_Common_Multiple/Largest_Common_Multiple_1-DArray.cpp
/*This code is to find the Largest__common_multiple of an given array of elements*/ 

#include <iostream>

using namespace std;

int getLCM(int a, int b) 
{
    int m;
    m = (a > b) ? a : b;
    while (true) {
        if (m % a == 0 && m % b == 0)
            // returns the value to getLCMArray function
            return m;
        m++;
    }
}

int getLCMArray(int arr[], int n) 
{
    int lcm = getLCM(arr[0], arr[1]);
    for (int i = 2; i < n; i++) 
    {
        // calls function getLCM to compute the LCM of two numbers
        lcm = getLCM(lcm, arr[i]); 
    }
    // returns LCM to main.
    return lcm; 
}

int main() 
{
    int n, i;
    cout << "Enter the no. of elements of array\n";
    cin >> n;
    int a[n];
    cout << "Enter the elements of array\n";
    for (i = 0; i < n; i++) 
    {
        cin >> a[i];
    }
    cout << "LCM of array elements: " << getLCMArray(a, n);
}

/*Enter the no. of elements of array
5
Enter the elements of array
4
6
12
24
30
LCM of array elements: 120*/
